### ISSUE TITLE
vendir cannot access the file because it is being used by another process on windows

### ISSUE THREAD

**Issue by meier-christoph** [carvel-dev/vendir#68]
**What steps did you take:**

When you add more than 1 path to directories using the http plugin you get these errors on windows. 

```yaml
---
apiVersion: vendir.k14s.io/v1alpha1
kind: Config
directories:
  - path: vendor/one
    contents:
      - path: one
        http:
          url: https://raw.githubusercontent.com/vmware-tanzu/carvel-vendir/develop/examples/http/vendir.yml

  - path: vendor/two
    contents:
      - path: two
        http:
          url: https://raw.githubusercontent.com/vmware-tanzu/carvel-vendir/develop/examples/http/vendir.yml
```

I suspect that the tmp dir is not unique and/or is shared with both downloads that run in some goroutines in parallel but then they try to delete the dir (or parent dir).

**What happened:**

```
Error: Syncing directory 'vendor/two': Deleting tmp dir '.vendir-tmp': remove .vendir-tmp\incoming\vendir-http521295291: The process cannot access the file because it is being used by another process.
```

**Environment:**

- vendir version (execute `vendir --version`): v0.19.0
- OS (e.g. from `/etc/os-release`): Windows 10

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by cppforlife** [comment]
@meier-christoph did you have two "vendir sync" running in parallel? can you confirm that running a single "vendir sync" on your system is resulting with this error?

**Comment by cppforlife** [comment]
the actual problem looks to be that we are not closing temp file [0]. even though we are trying to remove it we still need to do tmpFile.Close beforehand. fixing this issue should fix the clean up issue.

[0] https://github.com/vmware-tanzu/carvel-vendir/blob/05250dd89612f3fe07d9d5667e03ba70117a1f5a/pkg/vendir/fetch/http/sync.go#L34-L44

**Comment by meier-christoph** [comment]
only a single `vendir sync`, nice find on the missing close, updated the pr #70

**Comment by meier-christoph** [comment]
Fixed by #70
